src/box/replication.cc

```diff
@@ -45,6 +45,9 @@
 
 uint32_t instance_id = REPLICA_ID_NIL;
 struct tt_uuid INSTANCE_UUID;
+struct tt_hostname *INSTANCE_NAME = NULL;
+struct tt_hostname INSTANCE_NAME_CFG;
```

Comment:
There are 2 different subjects in one comment.

**Cfg variable naming**

You are saying that all the replication variables are fetched form the config directly, hence `box.cfg.replication_anon` should be stored as `replication_anon` and `box.cfg.instance_name` should be stored as `replication_instance_name` (why not just `instance_name` then?).

I disagree with that. My reasoning is that most of the replication options (and outside of it too) can be applied immediately. Hence they don't need separate cfg-variable and state-variable. But the anon flag and instance name can not be applied right away. Their application takes time and might even fail. It is confusing when you have a variable like `replication_is_something` which is true, but this "something" is really not true yet. Or `replication_some_state` equals `state1`, but the real state stored in WAL is `state2`.

Not only this is confusing in the code, but that also makes `box.info` misleading. You will see `box.info.name` new value even before it is applied. In addition, in case of `replication_anon`, while its change true->false is in progress, other replicas might try to connect to this instance like it is a master. Because its ballot already reflects the not yet applied state.

Thus I intend to keep "cfg" suffix. It clearly states that the variable does not store the actual state. Unless you have more arguments why "cfg" should not be used.

However I can propose a compromise. If you don't like the variable names being too different from the cfg names, I can move "cfg" to the beginning or to the end of the variables. Like `replication_anon_cfg`, `INSTANCE_NAME_CFG`, or `cfg_replication_anon`, `CFG_INSTANCE_NAME`. Then the cfg name will be included into the variable name as is. I also see that Sergey P. already uses the prefix "cfg_" in his `bootstrap_leader` patch. I should probably do the same to keep it consistent.

**Variable for the current instance name**

I made it a pointer because I didn't want to add 64 bytes of global memory. Now it is just 8. As for `NULL` vs nil - this is not a problem. The `tt_hostname` methods work with `NULL` and nil equally. If you use `tt_hostname` methods to work with the names, you won't notice any difference.

I can change it to a non-pointer plain object if you want. Even though it won't change how `INSTANCE_NAME` is accessed anyway (as long as `tt_hostname` struct and its methods remain).

Reply:
Suggestion: Maybe remove these global variables `replication_cfg_is_anon` and `INSTANCE_NAME_CFG` altogether and access the config directly (via `cfg_get`) right in the applier code where we need them? `cfg_get` should be fast and the code accessing them is a slow path anyway. Then we wouldn't have this argument about naming.

Reply:
It won't be possible. `cfg_get()` returns the values which want to be applied now but might be not validated yet. If `box.cfg` with a new invalid name and some other params would yield while applying those other params, and then an applier would try to use `cfg_get(name)` or any other place in the code would try that, they would get an invalid name.

Reply:
* I renamed anon option to `cfg_replication_anon`;
* Instance name is now stored as a plain value in `INSTANCE_NAME`;
* Instance name cfg is renamed to `cfg_instance_name`;

The `cfg_` prefix is selected to be consistent with `cfg_bootstrap_leader_uri` and `cfg_bootstrap_leader_uuid` which were merged while this PR is in work.